ingestion/src/metadata/ingestion/api/workflow.py

```diff
@@ -226,3 +226,44 @@ def print_status(self) -> int:
         else:
             click.secho("Workflow finished successfully", fg="green", bold=True)
             return 0
+
+    def log_flow_status(self) -> int:
+        from prefect import get_run_logger
+
+        logger = get_run_logger()
+        logger.info("Source Status: %s", self.source.get_status().as_string())
+        if hasattr(self, "stage"):
+            logger.info("Stage Status: %s", self.stage.get_status().as_string())
+        if hasattr(self, "sink"):
+            logger.info("Sink Status: %s", self.sink.get_status().as_string())
+        if hasattr(self, "bulk_sink"):
+            logger.info("Bulk Sink Status: %s", self.bulk_sink.get_status().as_string())
+
+
+def prefect_ingestion(config: str):
```

Comment:
instead of putting this into workflow.py which supposed to be a generic framework that can run on different orchestrators can we move it out of this and make it a pluggable interface?

Reply:
Good call, an alternative would be writing this as a separate Prefect Collection and we can certainly do it this way 👍  

Is there any reason other than CLI ingestion that you use `click.echo` for logging the workflow status information? I included this logging method only to expose ingestion status information in the Prefect UI, which wouldn't be available due to usage of click here rather than logging it

I can absolutely understand the rationale for separating this logic into a separate library, but I assumed a single function to compose ingestion workflow could be used even in Airflow DAGs to reduce the boilerplate - the same code to compose a workflow is currently called in every ingestion DAG

I'm OOO for the rest of the week - I will think about how to move this logic out into a Prefect Collection next week- thanks for reviewing!

Reply:
Hi! yep, `click.echo` is just for coloring CLI. We could actually improve that and prepare a fancy_log for CLI and a generic logging method that receives a logger as an argument for other use cases.

Ideally, the Workflow should be agnostic to wherever it runs and just handle the chain of execution. I think we do not have any direct link between Airflow <--> Workflow other than in the Airflow REST APIs when building the DAGs.

Now that we want to support multiple orchestration tools, we might need to prepare a Workflow `runners` module with all these integrations, one runner possibly being the CLI for example.

Also, thanks for removing the unused imports. Looks like `pycln` is not doing the best job here 😅 

We can catch up whenever you are available and plan this out!

Thanks!